history/history.c

```diff
@@ -1469,9 +1465,10 @@ int history_update_state(struct history_ctx *ctx, rpmts ts, const char *cmdline)
     history_set_cookie(ctx, cookie);
     ctx->trans_id = trans_id;
 error:
-    if (rc)
+    if (rc) {
         sqlite3_exec(ctx->db, "ROLLBACK;", 0, 0, NULL);
-    else
+        safe_free(current_ids);
+    } else
```

Comment:
nit: better to add braces for else as well.

Reply:
Different coding style in that file.